packages/clay-multi-select/src/index.tsx

```diff
@@ -219,6 +219,10 @@ const ClayMultiSelect = React.forwardRef<HTMLDivElement, IProps>(
 
 			const {keyCode} = event;
 
+			if (keyCode === BACKSPACE_KEY && !sourceItems.length) {
```

Comment:
```suggestion
			if (keyCode === BACKSPACE_KEY && !inputValue) {
```

@julien I think we'd better check `inputValue`, so we will ignore it only when the input is empty. That seems to work better.